README.md

```diff
@@ -1,43 +1,470 @@
-[![codecov](https://codecov.io/gh/dwyl/elixir-auth-github/branch/master/graph/badge.svg)](https://codecov.io/gh/dwyl/elixir-auth-github)
+# `elixir-auth-github`
 
-ElixirAuthGithub is a module to help you with Github OAuth in Elixir and Phoenix.
+The _easiest_ way to add GitHub OAuth authentication
+to your Elixir/Phoenix Apps.
 
-[Check it out on hex.pm](https://hex.pm/packages/elixir_auth_github/0.1.0)
+[![Build Status](https://img.shields.io/travis/com/dwyl/elixir-auth-github/master?color=bright-green&style=flat-square)](https://travis-ci.org/dwyl/elixir-auth-github)
+[![codecov.io](https://img.shields.io/codecov/c/github/dwyl/elixir-auth-github/master.svg?style=flat-square)](http://codecov.io/github/dwyl/elixir-auth-github?branch=master)
+[![Hex.pm](https://img.shields.io/hexpm/v/elixir_auth_github?color=brightgreen&style=flat-square)](https://hex.pm/packages/elixir_auth_github)
+[![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat-square)](https://github.com/dwyl/elixir-auth-github/issues)
+[![HitCount](http://hits.dwyl.io/dwyl/elixir-auth-github.svg)](http://hits.dwyl.io/dwyl/elixir-auth-github)
 
-We created it because everyone at dwyl uses github (including our clients!) so github OAuth makes sense for our internal (and external) tools. As a result, there's no use reinventing the wheel every project, and by making it into a module we can help other people as well!
+</div>
 
-First, add `:elixir_auth_github` to your deps in your mix.exs, then run `mix deps.get` in your terminal.
+
+# _Why_? 🤷
+
+We needed a **_much_ simpler**
+and **_extensively_ documented** way
+to add "_**Sign-in** with **GitHub**_"
+capability to our Elixir App(s). <br />
+
+> We created this package
+because everyone [@dwyl](https://github.com/dwyl)
+uses GitHub so using GitHub OAuth makes sense
+for our internal (and external) tools.
+By making it into a well-documented and tested
+reusable module other people can benefit from it.
+
+# _What_? 💭
+
+
+An Elixir package that seamlessly handles
+GitHub OAuth Authentication/Authorization
+in as few steps as possible. <br />
+Following best practices for security & privacy
+and avoiding complexity
+by having sensible defaults for all settings.
+
+
+# _Who_? 👥
+
+This module is for people building apps using Elixir/Phoenix
+who want to ship the "Sign-in with GitHub" feature _faster_
+and more maintainably.
+
+It's targetted at _complete_ beginners
+with no prior experience/knowledge
+of auth "schemes" or "strategies". <br />
+Just follow the detailed instructions
+and you'll be up-and running in 5 minutes.
+
+
+# _How_? 💻
+
+Add GitHub Auth to your Elixir/Phoenix project
+by following these 5 simple steps:
+
+> If you get stuck setting up your App,
+checkout our working demo:
+https://github.com/dwyl/elixir-auth-github-demo <br />
+> The demo is deployed on Heroku:
+https://elixir-auth-github-demo.herokuapp.com/
```

Comment:
🙌 

![image](https://user-images.githubusercontent.com/4185328/73612157-feea9a00-45e0-11ea-9085-20e5b0e56aff.png)

Reply:
Thanks for testing it and sharing your screenshot! 👍